### ISSUE TITLE
Menu language association to a menu item of type "Menu Item Alias" gives blank page

### ISSUE THREAD

**Issue by andrepereiradasilva** [joomla/joomla-cms#9911]
#### Steps to reproduce the issue

In a multi language site, create a menu structure like this:

###### Menu fr-FR

```
- Menu item 1 fr-FR (type Menu Item Alias to Menu item 2 fr-FR)
--- Menu item 2 fr-FR
```

###### Menu en-GB

```
- Menu item 1 en-GB (type Article) - language association to "Menu item 1 fr-FR"
```

#### Expected result

If we are in "Menu item 1 en-GB" when we change the language to french we should go to "Menu item 2 fr-FR"

#### Actual result

Blank page with component not found message ... because it tries to go to "Menu item 1 fr-FR"  (`/fr/?Itemid=xxxxx`)

#### System information (as much as possible)

Joomla Staging

#### Additional comments

@infograf768 can you check this one?

I think this may be a Menu alias type problem, ie.e not really related to associations. 
If we do a direct access (`/fr/?Itemid=xxxxx`) to that type of menu we get also the component not found message.

**Comment by brianteeman** [comment]
Sounds similar to #7334<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9911">issues.joomla.org/joomla-cms/9911</a>.</sub>

**Comment by andrepereiradasilva** [comment]
is similar, but not exactly the same thing

**Comment by brianteeman** [comment]
I wonder if it related

On 14 April 2016 at 19:35, andrepereiradasilva <notifications@github.com>
wrote:

> is similar, but not exactly the same thing
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/9911#issuecomment-210091703>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by andrepereiradasilva** [comment]
The problem of `/?Itemid=xxxxx` links not working is because the application is trying to dispatch to render the component and then it can't identify the component because Menu items of `alias` and `external URL` don't have a component associated. So we get `Component not found`.

Here: https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/application/site.php#L230

And then here https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/application/site.php#L124-L127

**Comment by infograf768** [comment]
Hmm... I guess we have to disable the associations for
separator,alias,heading,url

**Comment by andrepereiradasilva** [comment]
no i think is not needed.
The thing is Joomla should identify this menutypes and redirect to where it should.

**Comment by infograf768** [comment]
well, when you have a menuitem field (to redirect in login module for example), these are greyed.

**Comment by infograf768** [comment]
I have a very simple patch which will grey them and forbid to choose them when setting associations when ediing a menu.
Will do now.

**Comment by andrepereiradasilva** [comment]
this IMHO is not an associations problem but a routing problem.

**Comment by infograf768** [comment]
@andrepereiradasilva 
It will not prevent you from looking at another solution for alias, but definitely url, separator and heading should not be presented in the lists

**Comment by andrepereiradasilva** [comment]
ok

**Comment by andrepereiradasilva** [comment]
I guess for alias menu type something like this would work.

Replacing https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/application/site.php#L230, for this:
```php
// Special case for menu item types without components
$menu = $this->getMenu();
$activeMenu = $menu->getActive();

if (isset($activeMenu->type) && $activeMenu->type == 'alias')
{
	$aliasId = $activeMenu->params->get('aliasoptions', 0, 'int');
	$newMenuItem = $menu->getItem($aliasId);
	$this->redirect(JRoute::_($newMenuItem->link . '&Itemid=' . $aliasId));
}

// Dispatch the application
$this->dispatch();
```

This way all the `/?Itemid=xxxxx` menu type alias would redirect automatically to the new URL.

But i don't know if it would have other unintended consequences or even if this is the right way to do it in Joomla.

Of course if the menu item alias point to another menu item alias you would still have problems. A more robust solution would be needed.

**Comment by infograf768** [comment]
here is the patch to test. https://github.com/joomla/joomla-cms/pull/9929
Thanks.

**Comment by brianteeman** [comment]
Closed please test #9929<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9911">issues.joomla.org/joomla-cms/9911</a>.</sub>
